gadget-container/gadgets/networkpolicyadvisor/main.go

```diff
@@ -158,18 +154,61 @@ func main() {
 	}
 
 	// Start the BPF tracer
-	t, err := tracer.NewTracer(&tcpEventTracer{clientset})
+	mytracer := &tcpEventTracer{
+		clientset: clientset,
+		queue:     make(chan tracer.TcpV4, 500),
+	}
+	t, err := tracer.NewTracer(mytracer)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "%v\n", err)
 		os.Exit(1)
 	}
 
+	ticker := time.NewTicker(time.Second)
+	done := make(chan bool)
+	go func() {
+		for {
+			select {
+			case <-done:
+				return
+			case <-ticker.C:
+				eventCount := len(mytracer.queue)
+				if eventCount == 0 {
+					continue
+				}
+				// Consume that amount of events from the queue and use the same cache of
+				// pods and services with them. We might not consume all the events,
+				// that's ok, we'll get them at the next tick.
+				batch := make([]tracer.TcpV4, 0)
+				for i := 0; i < eventCount; i++ {
+					batch = append(batch, <-mytracer.queue)
+				}
```

Comment:
Is it possible to create `batch` with a length of `eventCount` elements and then fill each one without appending? Or, would it also be possible to create the batch outside the loop to avoid creating it in it iteration?

Reply:
> Is it possible to create batch with a length of eventCount elements and then fill each one without appending? 

Yes, it should be possible. Do you want to give it a try?

> Or, would it also be possible to create the batch outside the loop to avoid creating it in it iteration?

You would have to be careful the value of `eventCount` will change at every iteration, so I think it's not worth it.

Reply:
Done in https://github.com/kinvolk/inspektor-gadget/pull/74/commits/1975c6cc1c58aa04ba1b1052f1121ab683367353.